util/util.go

```diff
@@ -163,13 +162,13 @@ func AddImageNames(store storage.Store, image *storage.Image, addNames []string)
 func GetFailureCause(err, defaultError error) error {
 	switch nErr := errors.Cause(err).(type) {
 	case errcode.Errors:
-		return cli.NewMultiError([]error(nErr)...)
+		if strings.Contains(nErr.Error(), "not authorized") {
```

Comment:
The `errcode` package has a public `ErrorCodeUnauthorized` variable; it seems to me that this should be possible to do using the actual API instead of matching strings.

… if `not authorized` is even a value which can be returned here: AFAICS this is intended to match the code before https://github.com/containers/image/pull/410 ; so it definitely can’t apply in the `errcode.Errors` case, and could probably be dropped from the `default` case as well if `containers/image` is new enough.

Reply:
Thanks @mtrmac.  I've removed the hacks from both here and the default.  On a bad auth, we're hitting the first case and showing an appropriate error now rather than dropping down into the default.